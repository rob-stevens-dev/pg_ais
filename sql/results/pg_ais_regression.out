-- Load the extension
CREATE EXTENSION IF NOT EXISTS pg_ais;
ERROR:  could not load library "/usr/lib/postgresql/15/lib/pg_ais.so": /usr/lib/postgresql/15/lib/pg_ais.so: undefined symbol: ais_to_cstring
-- Regression test: clean table
DROP TABLE IF EXISTS test_ais;
NOTICE:  table "test_ais" does not exist, skipping
CREATE TABLE test_ais (id serial, sentence ais);
ERROR:  type "ais" does not exist
LINE 1: CREATE TABLE test_ais (id serial, sentence ais);
                                                   ^
-- ✅ 1. Valid multi-part message (reassembled)
SELECT pg_ais_parse('!AIVDM,2,1,2,B,53aGowP000001@D;E@E=:qD00000,0*5C');
ERROR:  function pg_ais_parse(unknown) does not exist
LINE 1: SELECT pg_ais_parse('!AIVDM,2,1,2,B,53aGowP000001@D;E@E=:qD0...
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
SELECT pg_ais_parse('!AIVDM,2,2,2,B,00000000000,2*21');
ERROR:  function pg_ais_parse(unknown) does not exist
LINE 1: SELECT pg_ais_parse('!AIVDM,2,2,2,B,00000000000,2*21');
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
-- ✅ 2. Out-of-order message (discarded)
SELECT pg_ais_parse('!AIVDM,2,2,9,B,00000000000,2*27');  -- arrives second part first
ERROR:  function pg_ais_parse(unknown) does not exist
LINE 1: SELECT pg_ais_parse('!AIVDM,2,2,9,B,00000000000,2*27');
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
SELECT pg_ais_parse('!AIVDM,2,1,9,B,53aGowP000001@D;E@E=:qD00000,0*5C');
ERROR:  function pg_ais_parse(unknown) does not exist
LINE 1: SELECT pg_ais_parse('!AIVDM,2,1,9,B,53aGowP000001@D;E@E=:qD0...
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
-- ✅ 3. Incomplete message (only one part)
SELECT pg_ais_parse('!AIVDM,2,1,3,B,53aGowP000001@D;E@E=:qD00000,0*5C');
ERROR:  function pg_ais_parse(unknown) does not exist
LINE 1: SELECT pg_ais_parse('!AIVDM,2,1,3,B,53aGowP000001@D;E@E=:qD0...
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
-- Optionally reset fragment buffer
-- SELECT pg_ais_reset();
