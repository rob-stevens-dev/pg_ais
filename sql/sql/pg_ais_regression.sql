-- Load the extension
CREATE EXTENSION IF NOT EXISTS pg_ais;

-- Regression test: clean table
DROP TABLE IF EXISTS test_ais;
CREATE TABLE test_ais (id serial, sentence ais);

-- ✅ 1. Valid multi-part message (reassembled)
SELECT pg_ais_parse('!AIVDM,2,1,2,B,53aGowP000001@D;E@E=:qD00000,0*5C');
SELECT pg_ais_parse('!AIVDM,2,2,2,B,00000000000,2*21');

-- ✅ 2. Out-of-order message (discarded)
SELECT pg_ais_parse('!AIVDM,2,2,9,B,00000000000,2*27');  -- arrives second part first
SELECT pg_ais_parse('!AIVDM,2,1,9,B,53aGowP000001@D;E@E=:qD00000,0*5C');

-- ✅ 3. Incomplete message (only one part)
SELECT pg_ais_parse('!AIVDM,2,1,3,B,53aGowP000001@D;E@E=:qD00000,0*5C');

-- Optionally reset fragment buffer
-- SELECT pg_ais_reset();

SELECT pg_ais_point('!AIVDM,1,1,,A,15Muq60001G?tTpE>Gbk0?wN0<0,0*7D');

-- Test pg_ais_point returns WKB binary as hex
SELECT encode(pg_ais_point('!AIVDM,1,1,,A,15Muq60001G?tTpE>Gbk0?wN0<0,0*7D'), 'hex');

-- Validate round-trip parse/point
SELECT (pg_ais_point(...) IS NOT NULL) AS ok;