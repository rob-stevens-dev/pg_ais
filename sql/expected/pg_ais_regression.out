-- Load the extension
CREATE EXTENSION IF NOT EXISTS pg_ais;

-- Regression test: clean table
DROP TABLE IF EXISTS test_ais;
CREATE TABLE test_ais (id serial, sentence ais);

-- ✅ 1. Valid multi-part message (reassembled)
 pg_ais_parse
---------------------------------------------------------------
 {"mmsi":123456789,"lat":42.123456,"lon":-70.654321,"speed":14.2,"heading":82.5}
(1 row)

-- ✅ 2. Out-of-order message (discarded)
 pg_ais_parse
--------------
 
(1 row)

 pg_ais_parse
--------------
 
(1 row)

-- ✅ 3. Incomplete message (ignored)
 pg_ais_parse
--------------
 
(1 row)

-- ✅ 4. Point output
 pg_ais_point
---------------------------
(-63.056667,44.651943)
(1 row)

-- ✅ 5. WKB hex output for point
 encode
----------------------------------
0000000001404f4adf5c28f5c3c05fae147ae147ae1
(1 row)

-- ✅ 6. Round-trip parse-point validation
 ok
----
 t
(1 row)

 id | shipname 
----+----------
  1 | SEAWAY
(1 row)

 should_be_null 
----------------
 
(1 row)

     pg_ais_point     
-----------------------
 (-122.341133,37.769535)
(1 row)

 pg_ais_get_int_field 
-----------------------
        366053209
(1 row)

 pg_ais_get_int_field 
-----------------------
                511
(1 row)

 pg_ais_get_int_field 
-----------------------
                  0
(1 row)

 pg_ais_get_int_field 
-----------------------
 
(1 row)

 pg_ais_get_float_field 
-------------------------
                     0
(1 row)

 pg_ais_get_float_field 
-------------------------
                 360.0
(1 row)

 pg_ais_get_float_field 
-------------------------
 
(1 row)

 pg_ais_get_bool_field 
------------------------
 f
(1 row)

 pg_ais_get_bool_field 
------------------------
 t
(1 row)

 pg_ais_get_bool_field 
------------------------
 
(1 row)
